app/src/main/java/ca/dal/cs/athletemonitor/athletemonitor/RecordActivity.java

```diff
@@ -286,6 +311,31 @@ public void toggleRecordStatus(View v) {
         }
     }
 
+
+    private static void updateLocationData(final UserLocation userLocation){
+        final String username = userLocation.getUsername();
+        final FirebaseDatabase db = FirebaseDatabase.getInstance();
+        final DatabaseReference userLocationRef
+                = db.getReference("user_locations" );
+
+        userLocationRef.addListenerForSingleValueEvent(new ValueEventListener() {
+            @Override
+            public void onDataChange(DataSnapshot dataSnapshot) {
+                if (!dataSnapshot.hasChild(username)) {
+                    userLocationRef.child(username).setValue(username);
```

Comment:
`.setValue(userLocation);` 
right? this looks like it just sets user_locations/[username] to a single string value [username]